bazel/mkosi/mkosi_image.bzl

```diff
@@ -141,3 +141,18 @@ mkosi_image = rule(
     executable = False,
     toolchains = ["@constellation//bazel/mkosi:toolchain_type"],
 )
+
+# This rule packages the given sources for inclusion into a Constellation rootfs. All files in the
+# output tarball are owned by root:root, date back to the UNIX epoch, are world readable, and
+# world executable if and only if one executable bit is set on the source file. This is different
+# from pkg_tar, where the mode is a fixed octal or needs to be passed for each file individually.
+#
+# NOTE: this rule dereferences symbolic links and hard links to weed out sources of nondeterminism.
+def reproducible_tar(name, srcs = [], visibility = []):
```

Comment:
Should we put a notice here that this requires GNU `tar`? Maybe explicitly target that notice to MacOS users, so they know that they have to install package X to run this.